jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/impl/OracleSqlDialect.java

```diff
@@ -440,4 +437,19 @@ public String getStringLiteral(final String value) {
         return "'" + value.replace("'", "''") + "'";
     }
 
+    @Override
+    public String generatePushdownSql(ConnectionInformation connectionInformation, String columnDescription, String pushdownSql) {
+        ImportType importType = getContext().getImportType();
+        if (importType == ImportType.JDBC) {
+            return super.generatePushdownSql(connectionInformation, columnDescription, pushdownSql);
+        } else {
+            if ((importType != ImportType.ORA)) throw new AssertionError("OracleSqlDialect has wrong ImportType");
+            final StringBuilder oracleImportQuery = new StringBuilder();
+            oracleImportQuery.append("IMPORT FROM ORA AT ").append(connectionInformation.getOraConnectionName()).append(" ");
+            oracleImportQuery.append(connectionInformation.getCredentials());
+            oracleImportQuery.append(" STATEMENT '").append(pushdownSql.replace("'", "''")).append("'");
+            return oracleImportQuery.toString();
```

Comment:
This code is a duplication, isn't it? Maybe we can extract a common method to some util class?

Reply:
It is slightly different every time. You could extract the common parts into a method, but this method would have a lot of parameters (this would not improve readability). Alternatively you could use a Builder instead of raw strings. There are plans to do that when the Builder @AnastasiiaSergienko  and @redcatbear are implementing, is ready.